pkg/controller/controller_binding.go

```diff
@@ -491,15 +497,53 @@ func (c *controller) injectServiceBinding(binding *v1beta1.ServiceBinding, crede
 	return err
 }
 
-func renameCredentialsKey(binding *v1beta1.ServiceBinding, key string) string {
-	for _, t := range binding.Spec.SecretTransform {
-		if t.RenameKey != nil {
-			if key == t.RenameKey.From {
-				return t.RenameKey.To
+func (c *controller) transformCredentials(transforms []v1beta1.SecretTransform, credentials map[string]interface{}) error {
+	for _, t := range transforms {
+		switch {
+		case t.AddKey != nil:
+			var value interface{}
+			if t.AddKey.JSONPathExpression != nil {
+				result, err := evaluateJSONPath(*t.AddKey.JSONPathExpression, credentials)
+				if err != nil {
+					return err
+				}
+				value = result
+			} else if t.AddKey.StringValue != nil {
+				value = *t.AddKey.StringValue
+			} else {
+				value = t.AddKey.Value
+			}
+			credentials[t.AddKey.Key] = value
+		case t.RenameKey != nil:
+			credentials[t.RenameKey.To] = credentials[t.RenameKey.From]
```

Comment:
Do we want to warn/error when the From key doesn't exist?

Reply:
I've opened an issue for this (#1956) and I'll try to identify all problems like this in a further PR. Since the `RenameKeyTransform` has been in master for some time now, I'm guessing I don't need to fix this in this PR.

Reply:
Agreed. Thanks for creating an issue to track that! 👍